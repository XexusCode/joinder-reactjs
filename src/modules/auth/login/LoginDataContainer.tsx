import { useDispatch } from "react-redux";
import React, { useEffect, useState } from "react";
import { login } from "../authActions";
import { LoginViewForm } from "./LoginViewForm";
import Swal from "sweetalert2";
import { apiCaller } from "../../../helpers/apiCaller";
import { Helmet } from "react-helmet";

export const LoginDataContainer: React.FunctionComponent = () => {
  const dispatch = useDispatch();

  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  const handleSubmit = async (username: string, password: string) => {
    const respuesta = await apiCaller(
      `auth/signIn`,
      { username, password },
      "POST",
      false
    );
    if (respuesta.success) {
      Swal.fire("Success", respuesta.message + "\n", "success");
      localStorage.setItem("token", respuesta.data.accesstoken);
      dispatch(
        login({ uid: respuesta.data.uid, username: respuesta.data.username })
      );
    }
  };

  useEffect(() => {
    apiCaller("auth/renew", {}, "GET", true).then((response) => {
      if (response.success) {
        localStorage.setItem("token", response.data.accesstoken);
        dispatch(
          login({
            uid: response.data.uid,
            username: response.data.username,
          })
        );
      }
    });
  }, [dispatch]);

  return (
    <div>
      <LoginViewForm
        handleSubmit={handleSubmit}
        email={username}
        handleInputEmail={setUsername}
        handleInputPassword={setPassword}
        password={password}
      />

      <Helmet>
        <meta charSet="utf-8" />
        <title>Login | Joinder</title>
        <meta name="description" content="Login del usuario " />
      </Helmet>
    </div>
  );
};
