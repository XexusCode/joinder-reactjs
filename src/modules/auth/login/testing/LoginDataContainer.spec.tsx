import * as React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import history from "../../../../routing/history";
import { LoginDataContainer } from "../LoginDataContainer";
import { Provider } from "react-redux";
import { store } from "../../../../redux/store/store";

describe("LoginDataContainer tests", () => {
  it("should take a snapshot of <LoginDataContainer/>", function () {
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <LoginDataContainer />
        </Router>
      </Provider>
    );
    expect(asFragment).toMatchSnapshot();
  });
});
