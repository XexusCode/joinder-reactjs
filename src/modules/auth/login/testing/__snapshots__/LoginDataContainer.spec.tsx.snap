// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginDataContainer tests should take a snapshot of <LoginDataContainer/> 1`] = `[Function]`;
